src/ValidatorKeys.cpp

```diff
@@ -0,0 +1,227 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of validator-keys-tool:
+        https://github.com/ripple/validator-keys-tool
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ValidatorKeys.h>
+#include <ripple/basics/StringUtilities.h>
+#include <ripple/json/json_reader.h>
+#include <ripple/json/to_string.h>
+#include <ripple/protocol/HashPrefix.h>
+#include <ripple/protocol/Sign.h>
+#include <beast/core/detail/base64.hpp>
+#include <boost/filesystem.hpp>
+#include <fstream>
+
+namespace ripple {
+
+std::string
+ValidatorToken::toString () const
+{
+    Json::Value jv;
+    jv["validation_secret_key"] = strHex(secretKey.data(), secretKey.size());
+    jv["manifest"] = manifest;
+
+    return beast::detail::base64_encode(to_string(jv));
+}
+
+ValidatorKeys::ValidatorKeys (KeyType const& keyType)
+    : keyType_ (keyType)
+    , tokenSequence_ (0)
+    , revoked_ (false)
+{
+    std::tie (publicKey_, secretKey_) = generateKeyPair (
+        keyType_, randomSeed ());
+}
+
+ValidatorKeys::ValidatorKeys (
+    KeyType const& keyType,
+    SecretKey const& secretKey,
+    std::uint32_t tokenSequence,
+    bool revoked)
+    : keyType_ (keyType)
+    , secretKey_ (secretKey)
+    , tokenSequence_ (tokenSequence)
+    , revoked_ (revoked)
+{
+    publicKey_ = derivePublicKey(keyType_, secretKey_);
```

Comment:
I actually added the `revoked` field to give users an easy way to recover if they "accidentally" revoke the keys (without actually adding the revocation to rippled.cfg). They can simply set `revoked` to false in their validator keys file, and their previous `token_sequence` wouldn't have been lost.
I also don't want to tell the user that their keys are revoked, if they haven't generated a revocation. So even if they muck up the `token_sequence`, the tool will tell them if it can't generate a token, but it will only indicate that the keys are revoked if `revoked` is true.